Samples/AudioPreprocessing/WinUI/AudioPreprocessing/AudioPreprocessing/Model/PreprocessingModel.cs

```diff
@@ -0,0 +1,244 @@
+﻿using Microsoft.AI.MachineLearning;
+using Microsoft.AI.MachineLearning.Experimental;
+using NAudio.Wave;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Windows.Graphics.Imaging;
+using Windows.Media;
+using Operator = Microsoft.AI.MachineLearning.Experimental.LearningModelOperator;
+
+namespace AudioPreprocessing.Model
+{
+    public class PreprocessModel
+    {
+        const string MicrosoftExperimentalDomain = "com.microsoft.experimental";
+
+        public string AudioPath { get; set; }
+
+        public string MelSpecImagePath { get; set; }
+
+        public PreprocessModel()
+        {
+            AudioPath = "..\\tmp\\mel_spectrogram_file.jpg";
+            MelSpecImagePath = "..\\tmp\\mel_spectrogram_file.jpg";
+        }
+
+        public SoftwareBitmap GenerateMelSpectrogram(string audioPath)
+        {
+            var signal = GetSignalFromFile(audioPath);
```

Comment:
do you want to do any validation on "audioPath"? It's just a string right now, so there's no guarantee it's a valid file path.

Reply:
I've added a very small check that it's a .wav file.